lib/input_data_page.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:age_calculator/calculated_data_page.dart';
+import 'package:age_calculator/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'input_data_bloc.dart';
+import 'constants.dart';
+
+class InputDataPage extends StatefulWidget {
+  @override
+  State<InputDataPage> createState() => _InputDataPageState();
+}
+
+class _InputDataPageState extends State<InputDataPage> {
+  final InputDataBloc _bloc = InputDataBloc(InputDataInitialState());
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Column(
+        children: [
+          const SizedBox(height: 15),
+          BlocBuilder<InputDataBloc, InputDataStates>(
+              bloc: _bloc,
+              builder: (context, state) {
+                print('state $state');
+                if (state is InputDataInitialState) {
+                  return Column(children: [
+                    datePick(Constants.fromDateNameString,
+                        DateTime(Constants.defaultDate)),
+                    const SizedBox(height: 15),
+                    datePick(Constants.toDateNameString, state.lastDate),
+                  ]);
+                } else if (state is PickedDateState) {
+                  return Column(children: [
+                    datePick(
+                        Constants.fromDateNameString, state.dateOfBirth),
+                    const SizedBox(height: 15),
+                    datePick(Constants.toDateNameString, state.lastDate),
+                  ]);
+                } else {
+                  return const Text('Unknown error');
+                }
+              }),
+          const SizedBox(height: 15),
+          ElevatedButton(
+            onPressed: () {
+              if (_bloc.dateOfBirth != DateTime(0000)) {
+                _bloc.calculateAge();
+                Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                        builder: (_) => BlocProvider.value(
+                              value: _bloc,
+                              child: CalculatedDataPage(),
+                            )));
+              } else {
+                showPicker(Constants.fromDateNameString);
+              }
+            },
+            child: const Text('Calculate Age'),
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget datePick(String widgetName, DateTime pick) {
```

Comment:
Extract to separate widget

Reply:
Why?
[Click here](https://stackoverflow.com/questions/53234825/what-is-the-difference-between-functions-and-classes-to-create-reusable-widgets)

Reply:
Extracted